### PR TITLE
feat: Add deprecated shape remover

### PR DESCRIPTION
**Pull request by sichanyoo**
## Issue \#

https://github.com/awslabs/aws-sdk-swift/issues/1026

## Description of changes

- Adds model preprocessing that removes shapes with deprecated trait based on `since` property if earlier than cutoff date.
- Adds codegen tests for the shape remover

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/model/AWSDeprecatedShapeRemoverTests.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import software.amazon.smithy.aws.traits.protocols.RestJson1Trait
+
+class AWSDeprecatedShapeRemoverTests {
+    @Test
+    fun `Shape deprecated before the cutoff date based on deprecated trait's since field gets removed`() {
+        val context = setupTests("deprecated-shape-removal-test.smithy", "com.test#Example")
+        val contents = TestUtils.getModelFileContents("Example", "OperationWithDeprecatedInputMembersInput.swift", context.manifest)
+        contents.shouldSyntacticSanityCheck()
+        val removedContent = """
+    @available(*, deprecated, message: " API deprecated since 2024-09-01")
```
Comment (jbelkins) [inline codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/model/AWSDeprecatedShapeRemoverTests.kt:18]: Why is there a leading space in the message?  Even if not part of this PR, we should remove that

**Review by jbelkins — APPROVED** [review]

File: codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/model/AWSDeprecatedShapeRemoverTests.kt (hunk 1 above)
Comment (sichanyoo) [reply codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/model/AWSDeprecatedShapeRemoverTests.kt:18]: Good catch; fixed as part of https://github.com/smithy-lang/smithy-swift/pull/732.
